Indicators/LeastSquaresMovingAverage.cs

```diff
@@ -63,23 +75,31 @@ public LeastSquaresMovingAverage(int period)
         /// </returns>
         protected override decimal ComputeNextValue(IReadOnlyWindow<IndicatorDataPoint> window, IndicatorDataPoint input)
         {
-            // Until the windows is ready, the indicator returns the input value.
-            decimal output = input;
-            if (IsReady)
-            {
-                // Sort the windows by time, convert the observations ton double and transform it to a double array
-                double[] series = window
-                    .OrderBy(i => i.Time)
-                    .Select(i => Convert.ToDouble(i.Value))
-                    .ToArray<double>();
-                // Fit OLS
-                Tuple<double, double> ols = Fit.Line(x: t, y: series);
-                var alfa = (decimal)ols.Item1;
-                var beta = (decimal)ols.Item2;
-                // Make the projection.
-                output = alfa + beta * (Period);
-            }
-            return output;
+            // Until the window is ready, the indicator returns the input value.
+            if (!IsReady) return input;
+
+            // Sort the window by time, convert the observations to double and transform it to an array
+            var series = window
+                .OrderBy(i => i.Time)
+                .Select(i => Convert.ToDouble(i.Value))
+                .ToArray();
+            // Fit OLS
+            var ols = Fit.Line(x: t, y: series);
+            Intercept.Update(input.Time, (decimal)ols.Item1);
+            Slope.Update(input.Time, (decimal)ols.Item2);
+
+            // Calculate the fitted value corresponding to the input
+            return Intercept + Slope * Period;
```

Comment:
In my first try, Intercept and Slope were not window indicator, but then I used Intercept.Plus(Slope.Times(Period)), it was returning zeroes. Then I moved to window and use that expression. Finally, when you suggested reviewing the usage of window indicator, I re-access everything and come up with this solution.

I will wait for final reviews and squash some commits (if everyone agrees) that were mostly typos and variable renaming.